src/EditorFeatures/Core/Extensibility/NavigationBar/AbstractEditorNavigationBarItemService.cs

```diff
@@ -71,12 +71,12 @@ protected void NavigateToPosition(Workspace workspace, DocumentId documentId, in
             ITextVersion textVersion,
             CancellationToken cancellationToken)
         {
-            // If the item points to a location in this document, then just determine the current location
-            // of that item and go directly to it.
-            var navigationSpan = item.TryGetNavigationSpan(textVersion);
-            if (navigationSpan != null)
+            if (symbolItem.Location.InDocumentInfo != null)
             {
-                return Task.FromResult((document.Id, navigationSpan.Value.Start, 0));
+                // If the item points to a location in this document, then just determine the where that span currently
+                // is (in case recent edits have moved it) and navigate there.
+                var navigationSpan = item.GetCurrentNavigationSpan(textVersion, symbolItem.Location.InDocumentInfo.Value.navigationSpan);
+                return Task.FromResult((document.Id, navigationSpan.Start, 0));
```

Comment:
this is basically an inline of the code we had at https://github.com/dotnet/roslyn/pull/57606/files#diff-e9f36b9abd4578acdcba3b15977c2d9e3c87d2c019bf912ca97e4b76d4f52c4fL43 that was only used in one place.  No need for special casing that there when we already special case navigating to SymbolItems here.